### PR TITLE
Just some code smell cleanups to appease Sonar

### PR DESCRIPTION
**Pull request by Michael1993**
Proposed commit message:

```
Small fixes for Sonar (#640)

This commit contains small changes related to Sonar code analysis.

PR: #640
```

---
**PR checklist**

The following checklist shall help the PR's author, the reviewers and maintainers to ensure the quality of this project.
It is based on our contributors guidelines, especially the ["writing code" section](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.md#writing-code).
It shall help to check for completion of the listed points.
If a point does not apply to the given PR's changes, the corresponding entry can be simply marked as done. 

Documentation (general)
* [ ] There is documentation (Javadoc and site documentation; added or updated)
* [ ] There is implementation information to describe _why_ a non-obvious source code / solution got implemented
* [ ] Site documentation has its own `.adoc` file in the `docs` folder, e.g. `docs/report-entries.adoc`
* [ ] Site documentation in `.adoc` file references demo in `src/demo/java` instead of containing code blocks as text
* [ ] Only one sentence per line (especially in `.adoc` files)
* [ ] Javadoc uses formal style, while sites documentation may use informal style

Documentation (new extension)
* [ ] The `docs/docs-nav.yml` navigation has an entry for the new extension
* [ ] The `package-info.java` contains information about the new extension

Code
* [ ] Code adheres to code style, naming conventions etc.
* [ ] Successful tests cover all changes
* [ ] There are checks which validate correct / false usage / configuration of a functionality and there are tests to verify those checks
* [ ] Tests use [AssertJ](https://assertj.github.io/doc/) or our own [PioneerAssert](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.md#assertions) (which are based on AssertJ)

Contributing
* [ ] A prepared commit message exists
* [ ] The list of contributions inside `README.md` mentions the new contribution (real name optional)

### CODE REVIEW

File: src/main/java/org/junitpioneer/jupiter/cartesian/ArgumentSets.java [hunk 1]
```diff
@@ -149,7 +149,7 @@ public final <T> ArgumentSets argumentsForNextParameter(Stream<T> arguments) {
 	}
 
 	List<List<?>> getArguments() {
-		return argumentSets;
+		return parameterSets;
```
Comment (beatngu13) [inline src/main/java/org/junitpioneer/jupiter/cartesian/ArgumentSets.java:152]: I find `parameterSets` within `ArgumentSets` a bit confusing. Alternatively, you could use the name of this getter, i.e., name the field `arguments`.

File: src/main/java/org/junitpioneer/jupiter/json/JacksonNode.java [hunk 2]
```diff
@@ -39,7 +39,7 @@ public boolean isArray() {
 
 	@Override
 	public Stream<Node> elements() {
-		return StreamSupport.stream(node.spliterator(), false).map(node -> new JacksonNode(objectMapper, node));
+		return StreamSupport.stream(node.spliterator(), false).map(n -> new JacksonNode(objectMapper, n));
```
Comment (beatngu13) [inline src/main/java/org/junitpioneer/jupiter/json/JacksonNode.java:42]: Hmmm, is this really an improvement? `element` maybe?

File: src/test/java/org/junitpioneer/jupiter/json/NoJsonParserConfiguredExceptionTest.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+package org.junitpioneer.jupiter.json;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+
+class NoJsonParserConfiguredExceptionTest {
+
+	@Test
+	void shouldBeUnchecked() {
```
Comment (beatngu13) [inline src/test/java/org/junitpioneer/jupiter/json/NoJsonParserConfiguredExceptionTest.java:20]: Really worth a test? 🫣

Comment (Michael1993) [reply src/test/java/org/junitpioneer/jupiter/json/NoJsonParserConfiguredExceptionTest.java:20]: Not sure. What else could we do?

File: src/main/java/org/junitpioneer/jupiter/json/JacksonNode.java (hunk 2 above)
Comment (Michael1993) [reply src/main/java/org/junitpioneer/jupiter/json/JacksonNode.java:42]: 👍

File: src/main/java/org/junitpioneer/jupiter/cartesian/ArgumentSets.java (hunk 1 above)
Comment (Michael1993) [reply src/main/java/org/junitpioneer/jupiter/cartesian/ArgumentSets.java:152]: 👍

**Review by beatngu13 — APPROVED** [review]
Could you prepare a commit message? Then feel free to merge. 👍

**Review by Bukama — APPROVED** [review]
LGTM (even that the test is strange)

**Pull request merged**